haskell_101/codelab/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM haskell:latest
+FROM haskell:8
```

Comment:
Clarifying question: why don't we pull from the latest tag?

Reply:
I noticed that there is a deprecation warning on cabal when installing dependencies using `cabal install <pkg>` so I figured locking the version would make sure people using this repo for learning haskell would never have a failing docker build in case haskell:9 came out, for example

Reply:
That's perfect. Let's also add a comment for this so that when future changes upgrade tag we know to check that the warnings are gone. Thank you